common/quotas/limiter_test.go

```diff
@@ -149,6 +122,241 @@ func BenchmarkMultiStageRateLimiter1000Domains(b *testing.B) {
 	}
 }
 
+func TestDynamicRateLimiter_RepeatedCalls(t *testing.T) {
+	ttl := time.Second
+	mockTime := clock.NewMockedTimeSource()
+	// Use the number of times the function has been called as a counter and ensure that we make no extra calls
+	rps := 0
+	rpsFunc := func() float64 {
+		res := rps
+		rps = rps + 1
+		return float64(res)
+	}
+	limiter := NewDynamicRateLimiter(rpsFunc, WithTimeSource(mockTime), WithTTL(ttl))
+	assert.Equal(t, limiter.Limit(), rate.Limit(0))
+	assert.Equal(t, limiter.Limit(), rate.Limit(0))
+	mockTime.Advance(ttl - 1)
+	assert.Equal(t, limiter.Limit(), rate.Limit(0))
+	mockTime.Advance(time.Nanosecond)
```

Comment:
mentioned above, but yea - I'd just avoid touching the boundary / making an assertion about it (beyond "does not explode in a fiery ball of process death" if that's a concern).  it doesn't matter to the semantics.